pkg/controller/controller_utils.go

```diff
@@ -832,7 +838,13 @@ func (s ActivePodsWithRanks) Less(i, j int) bool {
 		return len(s.Pods[i].Spec.NodeName) == 0
 	}
 	// 2. PodPending < PodUnknown < PodRunning
-	if podPhaseToOrdinal[s.Pods[i].Status.Phase] != podPhaseToOrdinal[s.Pods[j].Status.Phase] {
+	if _, ok := podPhaseToOrdinal[s.Pods[i].Status.Phase]; !ok {
+		klog.Errorf("phase %v is not in map", s.Pods[i].Status.Phase)
+	}
+	if _, ok := podPhaseToOrdinal[s.Pods[j].Status.Phase]; !ok {
+		klog.Errorf("the phase %v is not in map", s.Pods[i].Status.Phase)
```

Comment:
Should be `s.Pods[j].Status.Phase`.  But I'm not sure whether logging an error is the correct behavior.

Reply:
I am watching e2e test to see if any such log shows up.